test/ng/parseSpec.js

```diff
@@ -3900,6 +3900,34 @@ describe('parser', function() {
           expect(isFunction(s.toString)).toBe(true);
           expect(l.toString).toBe(1);
         }));
+
+        it('should assign a property to locals properties with null or undefined value', inject(function($parse) {
+          var local = {a:null};
+          $parse("a.b =1")(local);
+          expect(local.a).toEqual({b:1});
+
+          local.a = undefined;
+          $parse("a.b =1")(local);
+          expect(local.a).toEqual({b:1});
+        }));
+
+        it('should not assign a property to locals properties with falsy value', inject(function($parse) {
+          var local = {a:0};
+
+          expect(function() {
+            $parse("a.b =1")(local);
+          }).toThrow();
```

Comment:
I would also check that `local.a` is still the original value after each `.toThrow()`.